common/src/main/java/tachyon/util/network/NetworkAddressUtils.java

```diff
@@ -48,6 +50,141 @@
   private NetworkAddressUtils() {}
 
   /**
+   * Different types of services that client uses to connect. These types also indicate the service
+   * bind address
+   */
+  public enum ServiceType {
+
+    /**
+     * Master RPC service (Thrift)
+     */
+    MASTER_RPC("Tachyon Master RPC service", Constants.MASTER_HOSTNAME, Constants.MASTER_BIND_HOST,
+        Constants.MASTER_PORT, Constants.DEFAULT_MASTER_PORT),
+
+    /**
+     * Master web service (Jetty)
+     */
+    MASTER_WEB("Tachyon Master Web service", Constants.MASTER_WEB_HOSTNAME,
+        Constants.MASTER_WEB_BIND_HOST, Constants.MASTER_WEB_PORT,
+        Constants.DEFAULT_MASTER_WEB_PORT),
+
+    /**
+     * Worker RPC service (Thrift)
+     */
+    WORKER_RPC("Tachyon Worker RPC service", Constants.WORKER_HOSTNAME, Constants.WORKER_BIND_HOST,
+        Constants.WORKER_PORT, Constants.DEFAULT_WORKER_PORT),
+
+    /**
+     * Worker data service (Netty)
+     */
+    WORKER_DATA("Tachyon Worker data service", Constants.WORKER_DATA_HOSTNAME,
+        Constants.WORKER_DATA_BIND_HOST, Constants.WORKER_DATA_PORT,
+        Constants.DEFAULT_WORKER_DATA_PORT),
+
+    /**
+     * Worker web service (Jetty)
+     */
+    WORKER_WEB("Tachyon Worker Web service", Constants.WORKER_WEB_HOSTNAME,
+        Constants.WORKER_WEB_BIND_HOST, Constants.WORKER_WEB_PORT,
+        Constants.DEFAULT_WORKER_WEB_PORT);
+
+    // service name
+    public final String mServiceName;
+
+    // the key of connect hostname
+    public final String mHostNameKey;
+
+    // the key of bind hostname
+    public final String mBindHostKey;
+
+    // the key of port
+    public final String mPortKey;
+
+    // default port number
+    public final int mDefaultPort;
+
+    ServiceType(String serviceName, String hostNameKey, String bindHostKey, String portKey,
+        int defaultPort) {
+      this.mServiceName = serviceName;
+      this.mHostNameKey = hostNameKey;
+      this.mBindHostKey = bindHostKey;
+      this.mPortKey = portKey;
+      this.mDefaultPort = defaultPort;
+    }
+  }
+
+  /**
+   * Gets service connection hostname. If the connection hostname is not explicitly specified,
+   * Tachyon will try bind hostname. If the bind hostname is wildcard, Tachyon will automatically
+   * select an appropriate local hostname.
+   *
+   * @param service Service type used to connect
+   * @param conf Tachyon configuration used to look up the host resolution timeout
+   * @return the connection hostname that a client can use to reach the service.
+   */
+  public static String getConnectHost(ServiceType service, TachyonConf conf) {
```

Comment:
It is not clear to me what it means by "connection hostname", e.g., what's the difference between "connection hostname" and "hostname". Could you elaborate it here?